lib/pure/times.nim

```diff
@@ -526,9 +554,58 @@ when not defined(JS):
     var cTimeInfo = timeInfo # for C++ we have to make a copy,
     # because the header of mktime is broken in my version of libc
     result = mktime(timeInfoToTM(cTimeInfo))
+    when defined(windows):
+      var secondOffset: TimeImpl = 0
+      if result == Time(-1) and cTimeInfo.year <= 1970:
```

Comment:
This code seems to make a lot of assumptions with no references, can you add references to documents which describe what you implemented?

Also, this needs to be tested, thoroughly.